### PR TITLE
mpq/stormlib now works on windows

### PR DESCRIPTION
**Pull request by thegtproject**

### CODE REVIEW

File: formats/mpq/mpq.go [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 
-// #cgo LDFLAGS: -L./stormlib/ -lstorm -lz -lbz2 -lstdc++
-// #include <stdlib.h>
+// #ifdef _MPQ_WINDOWS
+// 		#include <windows.h>
+// 		#include <stdlib.h>
+// #endif
+// #ifdef _MPQ_LINUX
+//		#include <stdlib.h>
+//		#define WINAPI
```
Comment (FooSoft) [inline formats/mpq/mpq.go:9]: Should probably move all the `#define`s (except `bool`) here. Windows should already have them all defined, so they are only necessary on Linux.

File: formats/mpq/mpq.go [hunk 2]
```diff
@@ -65,7 +72,7 @@ func (f *file) Read(data []byte) (int, error) {
 	var bytesRead int
 	if result := C.SFileReadFile(f.handle, unsafe.Pointer(&data[0]), C.uint(len(data)), (*C.uint)(unsafe.Pointer(&bytesRead)), nil); result == 0 {
 		lastError := getLastError()
-		if lastError == 1002 { // ERROR_HANDLE_EOF
+		if lastError == 1002 || lastError == 38 { // ERROR_HANDLE_EOF
```
Comment (FooSoft) [inline formats/mpq/mpq.go]: This needs to be conditional based on platform. Who knows what error `38` is on Linux and error `1002` on Windows?

**Review by FooSoft — CHANGES_REQUESTED** [review]
Aside from those two nitpicks, looks pretty solid -- good work.

File: formats/mpq/build.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package mpq
+
+/*
+
+// Build Tags
+
+// Windows
+#cgo windows CFLAGS: -D_MPQ_WINDOWS
+#cgo windows LDFLAGS: -Lstormlib -lstorm -lwininet -lz -lbz2 -lstdc++
```
Comment (FooSoft) [inline formats/mpq/build.go:9]: LOL, stormlib requires wininet on Windows? Talk about feature creep.

File: formats/mpq/mpq.go (hunk 2 above)
Comment (thegtproject) [reply formats/mpq/mpq.go]: of course, i was just making the testing/tool/sample work, not correct :)   But i'll make this more elegant just for you

File: formats/mpq/build.go (hunk 3 above)
Comment (thegtproject) [reply formats/mpq/build.go:9]: yup 😏

File: formats/mpq/mpq.go (hunk 1 above)
Comment (thegtproject) [reply formats/mpq/mpq.go:9]: Gave it a shot, compiler complained about missing every one of them, so it seems both platforms need the defines?

File: formats/mpq/mpq.go (hunk 2 above)
Comment (thegtproject) [reply formats/mpq/mpq.go]: Added platform dependent "sysEOF" constant

**Comment by thegtproject** [comment]
changes implemented

**Comment by FooSoft** [comment]
Awesome, thanks for the changes!

**Pull request merged**